packages/aws/src/RPA/Cloud/AWS/__init__.py

```diff
@@ -116,17 +123,16 @@ def _init_client(
             or aws_key is None
             or aws_key.strip() == ""
         ):
-            raise KeyError(
-                "AWS key ID and secret access key are required "
-                " to use AWS cloud service: %s" % service_name
-            )
+            auth_params = {}
+        else:
+            auth_params = {
+                "aws_access_key_id": aws_key_id,
+                "aws_secret_access_key": aws_key,
+            }
+        if session_token:
+            auth_params["aws_session_token"] = session_token
```

Comment:
Q: Just to be sure, will the session token take precedence (when read by the server) in front of the key ID & access key?

Reply:
Session token is in addition to key ID and access key when using temporary credentials, without it, auth will fail.